lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -6893,7 +6987,14 @@ void EmitMethodFld(ParseNode *pnode, Js::RegSlot callObjLocation, Js::PropertyId
     bool isScoped = (byteCodeGenerator->GetFlags() & fscrEval) != 0 ||
         (isRoot && callObjLocation != ByteCodeGenerator::RootObjectRegister);
 
-    EmitMethodFld(isRoot, isScoped, pnode->location, callObjLocation, propertyId, byteCodeGenerator, funcInfo, registerCacheIdForCall);
```

Comment:
Okay. This must be in the knopName case of EmitCallTarget. EmitLoadInstance gives us the global object as the owning instance. Can you fix that in EmitLoadInstance (i.e., set instLocation to Js::Constants::NoRegister in the import/export case)? That should cause us to load the target the right way via Emit called on the knopName.

--Paul

From: Taylor Woll [mailto:notifications@github.com]
Sent: Monday, March 7, 2016 3:07 PM
To: Microsoft/ChakraCore <ChakraCore@noreply.github.com>
Cc: Paul Leathers <pleath@microsoft.com>
Subject: Re: [ChakraCore] Implement interpreter support for module import/exports (#457)


In lib/Runtime/ByteCode/ByteCodeEmitter.cpp<https://github.com/Microsoft/ChakraCore/pull/457#discussion_r55289771>:

> @@ -6893,7 +6987,14 @@ void EmitMethodFld(ParseNode *pnode, Js::RegSlot callObjLocation, Js::PropertyId

>      bool isScoped = (byteCodeGenerator->GetFlags() & fscrEval) != 0 ||

>          (isRoot && callObjLocation != ByteCodeGenerator::RootObjectRegister);

>

> -    EmitMethodFld(isRoot, isScoped, pnode->location, callObjLocation, propertyId, byteCodeGenerator, funcInfo, registerCacheIdForCall);

The stack isn't very big. Inside EmitCallTarget, we have a direct call to EmitMethodFld which is how we arrived here. Stack looks like this (hope formatting doesn't look terrible):

Chakra.dll!EmitMethodFld(ParseNode * pnode, unsigned int callObjLocation, int propertyId, ByteCodeGenerator * byteCodeGenerator, FuncInfo * funcInfo, bool registerCacheIdForCall) Line 6992    C++

Chakra.dll!EmitCallTarget(ParseNode * pnodeTarget, int fSideEffectArgs, unsigned int * thisLocation, unsigned int * callObjLocation, ByteCodeGenerator * byteCodeGenerator, FuncInfo * funcInfo) Line 7256  C++

Chakra.dll!EmitCall(ParseNode * pnode, unsigned int rhsLocation, ByteCodeGenerator * byteCodeGenerator, FuncInfo * funcInfo, int fReturnValue, int fEvaluateComponents, int fHasNewTarget, unsigned int overrideThisLocation) Line 7573 C++

Chakra.dll!Emit(ParseNode * pnode, ByteCodeGenerator * byteCodeGenerator, FuncInfo * funcInfo, int fReturnValue, bool isConstructorCall, ParseNode * bindPnode) Line 9881   C++

Chakra.dll!ByteCodeGenerator::EmitTopLevelStatement(ParseNode * stmt, FuncInfo * funcInfo, int fReturnValue) Line 956   C++

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/ChakraCore/pull/457/files#r55289771>.